[ci] Resolve Type Check failure blocking PR #703

<!-- codex-handoff-key-ci-pr-703 -->

## Why / Context
PR #703 is blocked by failing CI checks that were not auto-remediated by `ci_fixer.mjs`.
- PR: https://github.com/Solders-Girdles/GPT-Trader/pull/703
- Branch: `codex/692-enhancement-add-optimize-compare-matrix-`
- Handoff reason: no auto-fix mapping for current failing checks

## Required Outcome
- Fix the CI failure(s) on the PR branch with the smallest safe change.
- Keep all required checks green.
- Avoid unrelated refactors.

## Failing Checks
- Type Check — https://github.com/Solders-Girdles/GPT-Trader/actions/runs/21919859186/job/63296815602

## Evidence
- Type Check: Type-checking failure detected.
  - Type Check Type Check (MyPy) 2026-02-11T19:28:28.7096823Z src/gpt_trader/cli/commands/optimize/formatters.py:356: error: Incompatible types in assignment (expression has type "list[str]", variable has...
  - Type Check Type Check (MyPy) 2026-02-11T19:28:28.7099360Z src/gpt_trader/cli/commands/optimize/formatters.py:358: error: Invalid index type "int" for "dict[str, Any]"; expected type "str" [index]

## File Pointers
- `src/gpt_trader/cli/commands/optimize/formatters.py`

## Verification
- `gh pr checkout 703 --repo Solders-Girdles/GPT-Trader`
- Run the failing check(s) locally and confirm pass.
- Confirm required PR checks are green.

--- a/src/gpt_trader/cli/commands/optimize/formatters.py
+++ b/src/gpt_trader/cli/commands/optimize/formatters.py
@@ -352,9 +352,7 @@ def _format_comparison_matrix(
     col_widths: list[int] = []
     for col_idx in range(len(header)):
         max_width = max(
-            len(row_cells[col_idx])
-            for row_cells in all_rows
-            if len(row_cells) > col_idx
+            len(row_cells[col_idx]) for row_cells in all_rows if len(row_cells) > col_idx
         )
         col_widths.append(max_width)
 

